File: backend/app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Notifications\Notifiable;

class Customer extends Model
{
    use HasFactory, Notifiable;

    protected $table = 'tbl_customers';
    protected $primaryKey = 'customer_id';
    protected $fillable = [
        'name',
        'contact',
        'address',
        'email',
        'is_deleted',
    ];

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class, 'customer_id', 'customer_id');
    }
}

File: mobile/lib/features/customer/domain/customer.dart
class Customer {
  Customer({
    required this.id,
    required this.name,
    required this.contact,
    required this.address,
    this.email,
  });

  final int id;
  final String name;
  final String contact;
  final String address;
  final String? email;

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      id: (json['customer_id'] as num).toInt(),
      name: json['name'] as String? ?? '',
      contact: json['contact'] as String? ?? '',
      address: json['address'] as String? ?? '',
      email: json['email'] as String?,
    );
  }
}
